c/tskit/tables.c

```diff
@@ -5388,6 +5381,638 @@ ancestor_mapper_run(ancestor_mapper_t *self)
+    bool parent_should_be_added = true;
+
+    if (num_edges > 0) {
+        for (j = 0; j < num_edges; j++) {
+
+            // Stop if the processed node's time exceeds the max time.
+            current_parent = input_edges->parent[j];
+            current_time = self->tables->nodes.time[current_parent];
+            if (current_time > self->max_time) {
+                goto out;
+            }
+
+            // Extract segment.
+            seg = ibd_finder_alloc_segment(self, input_edges->left[j],
+                input_edges->right[j], input_edges->child[j]);
+            // Create a SegmentList holding all of the sample segments descending from
+            // seg.
+            u = seg->node;
+            if (self->is_sample[u]) {
+                ret = ibd_finder_enqueue_segment(self, seg->left, seg->right, seg->node);
+                if (ret != 0) {
+                    goto out;
+                }
+            } else {
+                if (self->ancestor_map_head[u] != NULL) {
+                    s = self->ancestor_map_head[u];
+                    while (s != NULL) {
+                        if (seg->left > s->left) {
+                            intvl_l = seg->left;
+                        } else {
+                            intvl_l = s->left;
+                        }
+                        if (seg->right < s->right) {
+                            intvl_r = seg->right;
+                        } else {
+                            intvl_r = s->right;
+                        }
+                        // Add to the segment queue.
+                        if (intvl_r - intvl_l > 0) {
+                            ret = ibd_finder_enqueue_segment(
+                                self, intvl_l, intvl_r, s->node);
+                            if (ret != 0) {
+                                goto out;
+                            }
+                        }
+                        s = s->next;
+                    }
+                }
+            }
+
+            // Calculate new ibd segments descending from the current parent.
+            if (self->segment_queue_size > 0) {
+                ret = ibd_finder_calculate_ibd(self, current_parent);
+            }
+            if (ret != 0) {
+                goto out;
+            }
+
+            // For samples that appear in the parent column of the edge table
+            if (self->is_sample[current_parent] && parent_should_be_added) {
+                ret = ibd_finder_enqueue_segment(
+                    self, 0, self->sequence_length, current_parent);
+                if (ret != 0) {
+                    goto out;
+                }
+                ret = ibd_finder_calculate_ibd(self, current_parent);
+                if (ret != 0) {
+                    goto out;
+                }
+                parent_should_be_added = false;
+            }
+
+            // if (j != num_edges - 1 && input_edges->parent[j + 1] != current_parent) {
```

Comment:
This smells a bit - it's fine leaving commented out code when developing, but once something is ready to be merged it should be cleaned up, removing stuff like this.

Reply:
https://en.wikipedia.org/wiki/Code_smell